microbench/src/main/java/io/netty/microbench/handler/ssl/AbstractSslEngineBenchmark.java

```diff
@@ -0,0 +1,279 @@
+        }
+
+        abstract SslProvider sslProvider();
+
+        static SSLEngine configureEngine(SSLEngine engine, String cipher) {
+            engine.setEnabledProtocols(new String[]{ PROTOCOL_TLS_V1_2 });
+            engine.setEnabledCipherSuites(new String[]{ cipher });
+            return engine;
+        }
+    }
+
+    public enum BufferType {
+        HEAP {
+            @Override
+            ByteBuffer newBuffer(int size) {
+                return ByteBuffer.allocate(size);
+            }
+        },
+        DIRECT {
+            @Override
+            ByteBuffer newBuffer(int size) {
+                return ByteBuffer.allocateDirect(size);
+            }
+
+            @Override
+            void freeBuffer(ByteBuffer buffer) {
+                PlatformDependent.freeDirectBuffer(buffer);
+            }
+        };
+
+        abstract ByteBuffer newBuffer(int size);
+
+        void freeBuffer(ByteBuffer buffer) { }
+    }
+
+    @Param
+    public SslEngineProvider sslProvider;
+
+    @Param
+    public BufferType bufferType;
+
+    // Includes cipher required by HTTP/2
+    @Param({ "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256", "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256" })
+    public String cipher;
+
+    protected SSLEngine clientEngine;
+    protected SSLEngine serverEngine;
+
+    private ByteBuffer cTOs;
+    private ByteBuffer sTOc;
+    private ByteBuffer serverAppReadBuffer;
+    private ByteBuffer clientAppReadBuffer;
+    private ByteBuffer empty;
+
+    protected final void initEngines() {
+        clientEngine = newClientEngine();
+        serverEngine = newServerEngine();
+    }
+
+    protected final void destroyEngines() {
+        ReferenceCountUtil.release(clientEngine);
+        ReferenceCountUtil.release(serverEngine);
+    }
+
+    protected final void initHandshakeBuffers() {
+        cTOs = allocateBuffer(clientEngine.getSession().getPacketBufferSize());
+        sTOc = allocateBuffer(serverEngine.getSession().getPacketBufferSize());
+
+        serverAppReadBuffer = allocateBuffer(
+                serverEngine.getSession().getApplicationBufferSize());
+        clientAppReadBuffer = allocateBuffer(
+                clientEngine.getSession().getApplicationBufferSize());
+        empty = allocateBuffer(0);
+    }
+
+    protected final void destroyHandshakeBuffers() {
+        freeBuffer(cTOs);
+        freeBuffer(sTOc);
+        freeBuffer(serverAppReadBuffer);
+        freeBuffer(clientAppReadBuffer);
+        freeBuffer(empty);
+    }
+
+    protected final void doHandshake() throws SSLException {
+        clientEngine.beginHandshake();
+        serverEngine.beginHandshake();
+
+        SSLEngineResult clientResult;
+        SSLEngineResult serverResult;
+
+        boolean clientHandshakeFinished = false;
+        boolean serverHandshakeFinished = false;
+
+        do {
+            int cTOsPos = cTOs.position();
+            int sTOcPos = sTOc.position();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "sTOcPos". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1854)